File: backend/app/models/stats.py
from pydantic import BaseModel


class StatParSource(BaseModel):
    source: str
    total: int
    convertis: int
    taux_conversion: float


class StatFormationDemandee(BaseModel):
    formation_id: str | None
    formation_titre: str
    total: int


class StatsOut(BaseModel):
    total_leads: int
    total_convertis: int
    taux_conversion_global: float
    par_source: list[StatParSource]
    formations_les_plus_demandees: list[StatFormationDemandee]

File: frontend/src/types/stats.ts
export interface StatParSource {
  source: string
  total: number
  convertis: number
  taux_conversion: number
}

export interface StatFormationDemandee {
  formation_id: string | null
  formation_titre: string
  total: number
}

export interface Stats {
  total_leads: number
  total_convertis: number
  taux_conversion_global: number
  par_source: StatParSource[]
  formations_les_plus_demandees: StatFormationDemandee[]
}
